src/main/java/org/owasp/esapi/waf/internal/InterceptingHTTPServletRequest.java

```diff
@@ -171,18 +172,43 @@ public Enumeration getDictionaryParameterNames() {
 	private class RAFInputStream extends ServletInputStream {
 		
 		RandomAccessFile raf;
+		boolean isDone = false;
```

Comment:
Functional changes in this review to add new API from dependency updates.

May require scrutiny.   @xeno6696 @kwwall

Reply:
On the surface this seems like adding in some safeguards that we didn't have.  Not sure I see any issues here just yet.

Reply:
My understanding is that each request is a unique thread.  I'm sure we could cause havoc if we spun up multiple threads in an isolate JVM and started calling methods on a single instance.

That being said, I don't think that's how this class is used at runtime.  Each request should be on an isolated thread, and each thread gets a unique class reference/instance.  The only shared/static class attribute is `CHUNKED_BUFFER_SIZE`, which should be marked `final`.  At that point the only shared reference is read-only and should be as thread-safe as it needs to be for a deployed environment (from my current understanding).

I can make the CHUNKED_BUFFER_SIZE as final, in these changes if desired?

Reply:
This has been reverted while rolling back to javax.servlet-api 3.0.1
https://github.com/ESAPI/esapi-java-legacy/pull/683/commits/54c56d989393b1a0c02cb3ef9a428f5c11d7729f

Reply:
restored in update to 3.1.0
https://github.com/ESAPI/esapi-java-legacy/pull/683/commits/959d9e12817804a78e719b2dc42dc49194d43b41